gems/aws-sdk-core/lib/aws-sdk-core/credentials_provider_chain.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+module AWS::SDK::Core
+  class CredentialsProviderChain
+    def call(config)
+      providers.each do |method_name|
+        provider = send(method_name, config)
+        return provider if provider
+      end
+      nil
+    end
+
+    private
+
+    def providers
+      [
+        :stubbed_credentials,
```

Comment:
I'm not sure stubbed credentials should really be part of this chain.  To me thats a separate consideration